gateway/metrics/metrics.go

```diff
@@ -81,6 +82,15 @@ func BuildMetricsOptions() MetricOptions {
 		[]string{"method", "path", "status"},
 	)
 
+	startedCounter := prometheus.NewCounterVec(
```

Comment:
One more comment, using a standard prometheus metric will still result in lag on this metric and a possible race with the idler between scrapes

Reply:
Agreed, but it will be better than our current worst case.